feat: add Qwen Code support to Spec Kit

ahmet-cetinkaya opened the pull request:
## Summary
- Add comprehensive Qwen Code integration as fourth AI assistant option
- Extend CLI, documentation, and build pipeline to support Qwen
- Maintain backward compatibility with existing assistants (Claude, Gemini, Copilot)

## Changes Made
- **CLI Integration**: Added Qwen as option in `src/specify_cli/__init__.py` with proper help text and tool checking
- **Shell Scripts**: Extended `scripts/update-agent-context.sh` to support QWEN.md generation  
- **GitHub Actions**: Added Qwen package creation in release workflow with `.qwen/commands/*.md` structure
- **Documentation**: Updated README.md, CONTRIBUTING.md, and templates to include Qwen CLI references
- **Templates**: Extended plan template to reference Qwen in agent file updates

## Testing
- ✅ CLI help shows Qwen option: `--ai claude, gemini, copilot, or qwen`
- ✅ All commands function correctly with Qwen integration
- ✅ Check command runs without errors
- ✅ Follows established patterns from other AI assistants

## Implementation Details
- **Command Structure**: Uses `.qwen/commands/*.md` format
- **Context File**: Generates `QWEN.md` in project root
- **CLI Tool**: Expects `qwen` command availability
- **Installation URL**: References `https://github.com/QwenLM/qwen-code`

This extends Spec Kit's AI assistant ecosystem to include Qwen while maintaining full compatibility with existing workflows.

vjaykrsna commented:
Can't wait to get qwen code support as well

vjaykrsna commented:
@ahmet-cetinkaya is this complete?
because i tried ```uvx --from git+https://github.com/ahmet-cetinkaya/spec-kit#feature/add-qwen-support specify```
but i didn't see qwen code cli there
guide me if i did it wrong

ahmet-cetinkaya commented:
@vjaykrsna There’s a small mistake in your command — when specifying a branch you should use `@` instead of `#`. Try this:
```bash
uvx --refresh --from git+https://github.com/ahmet-cetinkaya/spec-kit@feature/add-qwen-support specify
```
This will let you see the new option.

However, since **spec-kit fetches its templates from releases**, the tool won’t fully work this way.
To properly run the feature before release, you’ll need to clone the branch locally:

```bash
git clone https://github.com/ahmet-cetinkaya/spec-kit -b feature/add-qwen-support
cd spec-kit

uv sync
uv run specify --help
```

vjaykrsna commented:
by default it fetch from github had to modify the function to get it to work after generating the files manually
hope they officially merge this soon

localden commented:
@ahmet-cetinkaya - last request, can you rev the CLI version and update the changelog with the addition of Qwen support?

ahmet-cetinkaya commented:
@localden I've addressed your last request:

1. Bumped the CLI version from 0.0.4 to 0.0.5 in pyproject.toml
2. Updated the CHANGELOG.md with the new version and added Qwen Code support as a new feature

The changes have been pushed to the branch. This should be ready to merge now.

localden commented:
Thank you for contribution, @ahmet-cetinkaya! 🥳

aemr3 commented:
hey @ahmet-cetinkaya, @localden there are some missing definitions to support qwen, please check my pr: #64

SeanPtest commented:
I tried this moments ago, didn't work.   

uvx --from git+https://github.com/github/spec-kit.git specify init spec-sample
## Selected AI assistant: qwen
Selected script type: sh
No matching release asset found for pattern: spec-kit-template-qwen-sh 
## ... ... ... spec kit template-qwen.sh.zip isn't listed, the others are.  
Initialization failed: 1   

I did a search for the missing zip file in the magic merge.   We can uhh select qwen tho, atleast.

ahmet-cetinkaya commented:
@aemr3 @SeanPtest - Fixed! PR #321 adds missing Qwen assets to releases. specify init --ai qwen will work after   merge. Thanks! 🚀

Mingholy commented:
#330 Changed md to toml so that the `FileCommandLoader` can load spec commands into qwen-code.
